lib/block.ml

```diff
@@ -117,16 +120,38 @@ let dump ppf ({ file; line; section; labels; contents; value; _ } as b) =
     Fmt.(Dump.list dump_string)
     contents dump_value value
 
-let pp_lines syntax =
+let pp_lines syntax t =
   let pp =
-    match syntax with Some Syntax.Cram -> Fmt.fmt "  %s" | _ -> Fmt.string
+    match syntax with
+    | Some Syntax.Cram -> Fmt.fmt "  %s"
+    | Some Syntax.Mli ->
+        let i = ref (t.column + 2) in
+        let s = ref Fmt.string in
+        while !i > 0 do
+          s := Fmt.prefix (Fmt.unit " ") !s;
+          i := !i - 1
+        done;
+        !s
```

Comment:
Is this the equivalent of `Fmt.fmt "%s%s" (String.make (t.column + 2) ' ')` ?

Reply:
No :/. if I write:

```ocaml
let pp_lines syntax t =
  let pp =
    match syntax with
    | Some Syntax.Cram -> Fmt.fmt "  %s"
    | Some Syntax.Mli -> Fmt.fmt "%s%s" (String.make (t.column + 2) ' ')
    | _ -> Fmt.string
  in
  Fmt.(list ~sep:(unit "\n") pp)
```
I get the error:
```
File "lib/block.ml", line 131, characters 40-72:
Error: This expression has type string but an expression was expected of type
         Format.formatter
```

Reply:
Yes, sorry, the `ppf` argument comes first.
Also, padding with space characters is something `Format` does natively:
```ocaml
fun ppf -> Fmt.fmt "%*s%s" ppf (t.column + 2) ""
```

Reply:
Apologies, but I declare defeat. I have never used the `Fmt` library before and find it hard to understand. I have struggled for many hours with `Fmt` inside mdx. Unless I'm given the exact diff to apply here that will compile, I won't be able to improve this particular bit of code.

Reply:
This diff works for me:
```dif
--- a/lib/block.ml
+++ b/lib/block.ml
@@ -128,14 +128,7 @@ let pp_lines syntax t =
   let pp =
     match syntax with
     | Some Syntax.Cram -> Fmt.fmt "  %s"
-    | Some Syntax.Mli ->
-        let i = ref (t.column + 2) in
-        let s = ref Fmt.string in
-        while !i > 0 do
-          s := Fmt.prefix (Fmt.unit " ") !s;
-          i := !i - 1
-        done;
-        !s
+    | Some Syntax.Mli -> fun ppf -> Fmt.fmt "%*s%s" ppf (t.column + 2) ""
     | _ -> Fmt.string
   in
   Fmt.(list ~sep:(unit "\n") pp)
```

Reply:
what a dope i am! i get it now. see https://github.com/realworldocaml/mdx/pull/242